### PR TITLE
ref(pyupgrade): more f-strings

### PR DESCRIPTION
**Pull request by joshuarli**
This is andersk/fstrings pyupgrade @ d02513d698ff7851077940ed065fc2a9409712a4, which lets us convert more format calls.

I really looked over all of them, seems to be fine. Ended up doing a good amount (but less than expected) manual touchups after.

This started as an investigation into some of the %-style formatting were converted into format calls and then were not able to be pyupgraded to f-strings, for example `RedisBuffer._make_key` which is definitely a regression due to how often it'll be called.

I started out by allowing all function calls like so:

```diff
diff --git a/pyupgrade/_main.py b/pyupgrade/_main.py
index 07cc250..7caccba 100644
--- a/pyupgrade/_main.py
+++ b/pyupgrade/_main.py
@@ -534,11 +534,7 @@ def _simple_arg(arg: ast.expr) -> bool:
         isinstance(arg, ast.Name) or
         (fix_fstrings_inline_strings and isinstance(arg, ast.Str)) or
         (isinstance(arg, ast.Attribute) and _simple_arg(arg.value)) or
-        (
-            isinstance(arg, ast.Call) and
-            _simple_arg(arg.func) and
-            not arg.args and not arg.keywords
-        )
+        isinstance(arg, ast.Call)
     ) 
```

...but before I started to look at `_unparse` to accomodate that, I noticed https://github.com/asottile/pyupgrade/pull/298 would work.

There are still a good amount of format calls left, but I think after this it's easier and quicker to just manually audit the rest of them rather than try and implement automated fixes... the biggest remaining category I see is like, `"{}".format(d["foo"])`, but it's not easy to get this right.

### CODE REVIEW

File: src/sentry/api/endpoints/project_users.py [hunk 1]
```diff
@@ -30,7 +30,7 @@ def get(self, request, project):
                 return Response([])
             try:
                 queryset = queryset.filter(
-                    **{"{}__icontains".format(EventUser.attr_from_keyword(pieces[0])): pieces[1]}
+                    **{f"{EventUser.attr_from_keyword(pieces[0])}__icontains": pieces[1]}
```
Comment (joshuarli) [inline src/sentry/api/endpoints/project_users.py:33]: The downside to automated f-string conversions is that changes like these aren't very readable anymore, and instead should be refactored to pull things out into more meaningful variable names.

Comment (joshuarli) [reply src/sentry/api/endpoints/project_users.py:33]: A more extreme example of this is what I mentioned in the PR description, https://github.com/getsentry/sentry/pull/23933/commits/35d4d8f753c0d2623467fd6d1018546351ff3e6a, which I refactored after an initial automated pass.

File: tests/sentry/api/endpoints/test_group_external_issue_details.py [hunk 2]
```diff
@@ -23,7 +23,7 @@ def test_deletes_external_issue(self):
         assert not PlatformExternalIssue.objects.filter(id=self.external_issue.id).exists()
 
     def test_handles_non_existing_external_issue(self):
-        url = "/api/0/issues/{}/external-issues/{}/".format(self.group.id, 99999)
+        url = f"/api/0/issues/{self.group.id}/external-issues/{99999}/"
```
Comment (joshuarli) [inline tests/sentry/api/endpoints/test_group_external_issue_details.py:26]: Ah, this one's interesting, I never implemented integer inlining.

Comment (joshuarli) [reply tests/sentry/api/endpoints/test_group_external_issue_details.py:26]:
```
$ rg -tpy 'f".*\{[0-9]+\}'

tests/snuba/api/endpoints/test_organization_events_v2.py
2523:        response = self.do_request({"field": f"count_at_least(measurements.fcp, {0})"})
```

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**